core/src/main/java/org/elasticsearch/ingest/core/CompoundProcessor.java

```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
+//TODO(simonw): can all these classes go into org.elasticsearch.ingest?
```

Comment:
I would be more in favour of keeping the two packages for now :) logic separation seems likes a good enough reason to keep two packages, that doesn't mean that the core package can already go out with no changes but the changes required should be small-ish, while I am afraid if we merge everything into a single package we will quickly lose this distinction (think of making IngestDocument and Pipeline writeable instead of having separate classes for that), that feels like undoing some of the work we have already done.